backend/django/create-model-and-api.mo.md

```diff
@@ -0,0 +1,219 @@
+# [MO] Create a model and an api route (~20 mins)
+
+## Owner: Alice Breton
+
+## Prerequisites (~2 mins)
+
+- Have a django project installed, [see this](https://github.com/bamlab/dev-standards/blob/master/backend/django/getting-started.mo.md).
+- Have an app (in this example Publications).
+- In this example we will use an already existing model 'city' in a 'locations' app, to show how to integrate other models with foreign keys.
+
+## Steps  
+
+- In this tutorial we will create a News model, add it to the Django admin back-office and an API route to get all the pieces of news that have been created.
+- To give a full example, we will ==== REMPLIR
+
+### Create a News model *(~10 min)*
+
+- In this example we will create a new model called News, with different attributes as you will see bellow. In the models.py file of your app (in our case Publications) create a new model. 
+
+ourproject/publications/models.py
+```python
+from django.db import models
+
+class News(models.Model):
+    title = models.CharField(max_length=50)   #CharField is a single line form in the admin; max_length speaks for itself
+    description = models.TextField(max_length=1000)   #TextField is a scrollable paragraph in the admin
+    created_at = models.DateTimeField(auto_now_add=True)    #auto_now_add generates automatically a timestamp when you create your create the news
+    updated_at = models.DateTimeField(auto_now=True)    #auto_now updates the timestamp everytime you change the news
+    city = models.ForeignKey('locations.city')    #All the cities of the City model of the locations app
+
+    class Meta:
+        verbose_name_plural = "News"    #The name of the class should be singular, this forces the plural to be News and not Newss
+
+    def __str__(self):
+        return self.title
+```
+
+
+- If you have users that are in groups, they need the permission to access to the Model. We won't explain how to create a group, we just want to show you how to add permissions to them. They will be added automaticcaly when you deploy.
+
+ouproject/users/apps/py
+```python
+from django.apps import AppConfig
+from django.contrib.admin import site
+from django.db.models.signals import post_migrate
+
+
+class UsersConfig(AppConfig):
+    name = 'intramuros.users'
+    verbose_name = "Users"
+
+    def ready(self):
+        post_migrate.connect(add_group_permissions, sender=self)
```

Comment:
A bit of "what" and "why" would be welcomed

Reply:
Maybe you could change the order of this code. 
- First explain add_model_permissions: It takes all the permission of a model and adds it to a group
- Then def add_group_permission: This function adds the News access right to the Group "Mayor Admin". 
- Then the function `ready` is called when the app is ready. What it does is hook a postMigrate method that will be ran after the migrations are done (wahou)